clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx

```diff
@@ -0,0 +1,208 @@
+   *
+   * Action will called from confirmation action-sheet.
+   */
+  const onCancelConfirmPress = React.useCallback(() => {
+    actionCancelRequest.onClose();
+    onLeavePress();
+  }, [actionCancelRequest, onLeavePress]);
+
+  /**
+   * On respond invitation button press.
+   *
+   * Need to show action-sheet containing two buttons:
+   * - Accept invitation
+   * - Reject invitation
+   * - Close
+   */
+  const onRespondInvitationPress = React.useCallback(() => {
+    actionRespondInvitation.onOpen();
+  }, [actionRespondInvitation]);
+
+  /**
+   * On respond invitation accept button press.
+   *
+   * Action will called from confirmation action-sheet.
+   */
+  const onInvitationAcceptPress = React.useCallback(() => {
+    actionRespondInvitation.onClose();
+    setJoinStatus(FanClubMemberStatusEnum.active);
+
+    fanClubMembersStore.setResourceId(data.id);
+    fanClubMembersStore.setUpdateFieldValue('type', 'accept-invitation');
+    fanClubMembersStore.updateData();
+  }, [actionRespondInvitation, data.id, fanClubMembersStore]);
+
+  /**
+   * On respond invitation reject button press.
+   *
+   * Action will called from confirmation action-sheet.
+   */
+  const onInvitationRejectPress = React.useCallback(() => {
+    actionRespondInvitation.onClose();
+    setJoinStatus(undefined);
+
+    fanClubMembersStore.setResourceId(data.id);
+    fanClubMembersStore.setUpdateFieldValue('type', 'reject-invitation');
+    fanClubMembersStore.updateData();
+  }, [actionRespondInvitation, data.id, fanClubMembersStore]);
+
+  React.useEffect(() => {
+    if (storeAdd.status === 'error') {
+      setJoinStatus(undefined);
+    }
+    if (storeDelete.status === 'error') {
+      setJoinStatus(
+        data.privacy === FanClubPrivacyEnum.public
+          ? FanClubMemberStatusEnum.active
+          : FanClubMemberStatusEnum.pendingRequest
+      );
+    }
+  }, [joinStatus, data.privacy, storeAdd.status, storeDelete.status]);
+
+  const { icon, button } = React.useMemo((): ButtonAttributeInterface => {
+    if (!joinStatus) {
+      return buildButtonAttributes(joinStatus, theme.colors, onJoinPress);
+    }
+
+    switch (joinStatus) {
+      case FanClubMemberStatusEnum.active:
+        return buildButtonAttributes(joinStatus, theme.colors, onLeavePress);
+      case FanClubMemberStatusEnum.pendingRequest:
+        return buildButtonAttributes(joinStatus, theme.colors, onCancelRequestPress);
+
+      case FanClubMemberStatusEnum.pendingInvitation:
+        return buildButtonAttributes(joinStatus, theme.colors, onRespondInvitationPress);
+
+      case FanClubMemberStatusEnum.banned:
+        return buildButtonAttributes(joinStatus, theme.colors);
+    }
+  }, [
+    onJoinPress,
+    onLeavePress,
+    onCancelRequestPress,
+    onRespondInvitationPress,
+    joinStatus,
+    theme.colors,
+  ]);
+
+  return (
+    <>
+      <Button
+        leftIcon={
+          !icon ? undefined : <Icon name={icon.name} color={icon.color} size="ic-xs" />
```

Comment:
isn't `icon && <Icon name={icon.name} color={icon.color} size="ic-xs" />` better?

Reply:
yeah, it's better :)